pkg/apis/v1beta1/ec2nodeclass_status.go

```diff
@@ -68,4 +68,7 @@ type EC2NodeClassStatus struct {
 	// InstanceProfile contains the resolved instance profile for the role
 	// +optional
 	InstanceProfile string `json:"instanceProfile,omitempty"`
+	// The generation observed by the nodeclass controller.
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
```

Comment:
This should be a per-status condition observedGeneration rather than a status-wide observed generation. See: https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/types.go#L1553. That also allows us to indicate "This thing has been updated for this generation (because it wasn't changed) but this other thing wasn't updated."

Reply:
Maybe just wrap the upstream status conditions library right now? Or somehow coordinate the move-over of the knative status conditions library to operatorpkg?